debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/structureformat/extensions/J9ObjectStructureFormatter.java

```diff
@@ -248,53 +263,111 @@ private void formatArrayObject(PrintStream out, J9ClassPointer localClazz, U8Poi
 	{
 		out.println(String.format("J9VMJavaLangString at %s {\n", localObject.getHexAddress()));
-		printJ9ObjectFields(out, tabLevel, localClazz, dataStart, localObject);
+		printJ9ObjectFields(out, tabLevel, localClazz, dataStart, localObject, address);
 		padding(out, tabLevel);
 		out.println(String.format("\"%s\"\n", J9ObjectHelper.stringValue(localObject)));
-		out.println("}");		
+		out.println("}");
 	}
 	
-	private void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ObjectPointer localObject) throws CorruptDataException 
+	public static void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ObjectPointer localObject, long address) throws CorruptDataException 
+	{
+		printJ9ObjectFields(out, tabLevel, localClazz, dataStart, localObject, address, null, false);
+	}
+	
+	public static void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ObjectPointer localObject, long address, String[] nestingHeirarchy, boolean showNestedFields) throws CorruptDataException 
 	{
-		J9ClassPointer instanceClass = localClazz;
 		long superclassIndex;
 		long depth;
 		J9ClassPointer previousSuperclass = J9ClassPointer.NULL;
 		boolean lockwordPrinted = false;
+		UDATA nestedFieldOffset = new UDATA(0);
+		boolean flatObject = false;
+		
+		J9ClassPointer nestedClassHeirarchy[] = null;
+		
+		if ((null != nestingHeirarchy) && ( nestingHeirarchy.length > 0)) {
+			flatObject = true;
+		}
+		
+		/* This is true when !flatobject is specified are currently printing a nested structure */ 
+		boolean printingEntireFlattenedStructureMember = showNestedFields && flatObject;
 		
 		if (J9BuildFlags.thr_lockNursery) {
 			lockwordPrinted = false;
 		}
 		
-		/* print individual fields */
-		J9UTF8Pointer classNameUTF = instanceClass.romClass().className();
-		padding(out, tabLevel);		
-		out.println(String.format("struct J9Class* clazz = !j9class 0x%X   // %s", localClazz.getAddress(), J9UTF8Helper.stringValue(classNameUTF)));
-		padding(out, tabLevel);
-		out.println(String.format("Object flags = %s;", J9ObjectHelper.flags(localObject).getHexValue()));
-		
-		if (!J9BuildFlags.thr_lockNursery) {
-			UDATA lockword = J9ObjectHelper.monitor(localObject);
-			if (lockword != null) {
+		if (flatObject) {
+			/* If it is a flatObject skip ahead to the nested field */
+			nestedClassHeirarchy = ValueTypeHelper.findNestedClassHeirarchy(localClazz, nestingHeirarchy);
+			J9ClassPointer clazz = null;
+			
+			U32 flags = new U32(J9VM_FIELD_OFFSET_WALK_INCLUDE_INSTANCE | J9VM_FIELD_OFFSET_WALK_INCLUDE_HIDDEN);
+			for (int i = 0; i < nestedClassHeirarchy.length - 1; i++) {
+				clazz = nestedClassHeirarchy[i];
+				depth = J9ClassHelper.classDepth(clazz).longValue();
+				boolean notFound = true;
```

Comment:
This would read better without the double negatives, i.e. `boolean found = false;`.